server/src/main/java/org/elasticsearch/action/bulk/TransportAbstractBulkAction.java

```diff
@@ -412,14 +413,43 @@ private void applyPipelinesAndDoInternalExecute(
         while (bulkRequestModifier.hasNext()) {
             req = bulkRequestModifier.next();
             i++;
+            doStreamsChecks(bulkRequest, projectMetadata, req, bulkRequestModifier, i);
+        }
+
+        var wrappedListener = bulkRequestModifier.wrapActionListenerIfNeeded(listener);
 
-            for (StreamType streamType : StreamType.getEnabledStreamTypesForProject(projectMetadata)) {
-                if (req instanceof IndexRequest ir && streamType.matchesStreamPrefix(req.index()) && ir.isPipelineResolved() == false) {
-                    IllegalArgumentException e = new IllegalArgumentException(
+        if (applyPipelines(task, bulkRequestModifier.getBulkRequest(), executor, wrappedListener) == false) {
+            doInternalExecute(task, bulkRequestModifier.getBulkRequest(), executor, wrappedListener, relativeStartTimeNanos);
+        }
+    }
+
+    private void doStreamsChecks(
+        BulkRequest bulkRequest,
+        ProjectMetadata projectMetadata,
+        DocWriteRequest<?> req,
+        BulkRequestModifier bulkRequestModifier,
+        int i
+    ) {
+        for (StreamType streamType : StreamType.getEnabledStreamTypesForProject(projectMetadata)) {
+            if (req instanceof IndexRequest ir && ir.isPipelineResolved() == false) {
+                IllegalArgumentException e = null;
+                if (streamType.matchesStreamPrefix(req.index())) {
+                    e = new IllegalArgumentException(
                         "Direct writes to child streams are prohibited. Index directly into the ["
                             + streamType.getStreamName()
                             + "] stream instead"
                     );
+                }
+
+                if (e == null && bulkRequest.streamsRestrictedParamsUsed() && req.index().equals(streamType.getStreamName())) {
```

Comment:
Two things about this:
(1) It would be useful if it also printed out which parameters were passed, because I had to debug in to see why my request had failed
(2) I think it is accidentally a little too restrictive. If I run this command:
```
POST logs/_doc
{
  "@timestamp": "2099-11-15T13:12:00",
  "message": "GET /search HTTP/1.1 200 1070000",
  "user": {
    "id": "kimchy"
  }
}
```
then I get rejected. Through debugging, I discovered that that is implicitly adding `pretty` as a parameter. I think we ought to allow that. If I use the form of that API that passes in an ID, that also gets blocked by this. Maybe we want to prevent users from providing their own IDs, but you can still pas an ID in the bulk API, so that seems inconsistent.

Reply:
Rethinking my last comment, I think there are 3 things that need to be done:
(1) List the request params in the error message
(2) Add `pretty` and `_id` to the whitelisted params
(3) Decide if we want to prevent users from setting the ID, and if so do it at a lower level so that it's consistent